### PR TITLE
Edit improvements

### PR DESCRIPTION
**Pull request by etokareva**

### CODE REVIEW

File: src/core/edit/edit.cell.view.js [hunk 1]
```diff
@@ -49,9 +52,17 @@ export class EditCellView {
 					}
 					else {
 						cell = model.navigation().cell;
+						const code = this.shortcut.keyCode;
+						if (code && code.length > 0) {
+							const reserved = Array.from(this.shortcut.codeMap.values()).some(val => val === code);
```
Comment (klumba12) [inline src/core/edit/edit.cell.view.js]: add to shortcut isControl method instead of that

File: src/core/edit/edit.cell.view.js [hunk 2]
```diff
@@ -49,9 +52,17 @@ export class EditCellView {
...
+							if (!reserved) {
+								this.memory = code;
```
Comment (klumba12) [inline src/core/edit/edit.cell.view.js]: why we need memory?

File: src/core/edit/edit.cell.view.js [hunk 3]
```diff
@@ -49,9 +52,17 @@ export class EditCellView {
 					}
 					else {
 						cell = model.navigation().cell;
+						const code = this.shortcut.keyCode;
+						if (code && code.length > 0) {
```
Comment (klumba12) [inline src/core/edit/edit.cell.view.js]: in what cases length is 0?

File: src/core/edit/edit.cell.view.js [hunk 4]
```diff
@@ -49,9 +52,17 @@ export class EditCellView {
...
+							const reserved = Array.from(this.shortcut.codeMap.values()).some(val => val === code);
+							if (!reserved) {
+								this.memory = code;
+								cell.value += code;
+							}
+						}
 					}
 
-					if (cell && model.edit().enter.execute(this.contextFactory(cell, cell.value, cell.label)) !== false) {
+					if (model.edit().enter.execute(this.contextFactory(cell, cell.value, cell.label)) !== false) {
```
Comment (klumba12) [inline src/core/edit/edit.cell.view.js]: why cell is excluded from if? it can be null

File: src/core/edit/edit.cell.view.js [hunk 5]
```diff
@@ -201,19 +221,30 @@ export class EditCellView {
...
+	}
+
 	commitShortcut() {
 		const model = this.model;
-		const commitShortcuts = model.edit().commitShortcuts;
+		const shortcuts = model.edit().commitShortcuts;
 		const cell = this.editor.cell || model.navigation().cell;
-		if (cell && commitShortcuts.hasOwnProperty(cell.column.type)) {
-			return commitShortcuts[cell.column.type];
+		if (cell && shortcuts.hasOwnProperty(cell.column.type)) {
```
Comment (klumba12) [inline src/core/edit/edit.cell.view.js]: please add editor check here

File: src/core/edit/edit.cell.view.js [hunk 6]
```diff
@@ -201,19 +221,30 @@ export class EditCellView {
...
+		return shortcuts['$default'];
 	}
 
-	get options() {
-		return this.editor.options;
+	enterShortcut() {
+		const model = this.model;
+		const shortcuts = model.edit().enterShortcuts;
+		const cell = this.editor.cell || model.navigation().cell;
+		if (cell && shortcuts.hasOwnProperty(cell.column.type)) {
```
Comment (klumba12) [inline src/core/edit/edit.cell.view.js]: please add editor check here

File: src/core/edit/edit.cell.view.js (hunk 2 above)
Comment (etokareva) [reply src/core/edit/edit.cell.view.js]: I need a flag to know about starting the input modifications with any key (to add it to the cell value, and to remove it on cancel)

File: src/core/behavior/shortcut.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+export class Shortcut {
+	constructor(manager) {
+		this.manager = manager;
+	}
+	get keyCode() {
+		return Shortcut.keyCode;
+	}
+
+	get isControl() {
+		return Array.from(codeMap.values()).some(val => val === Shortcut.keyCode);
```
Comment (klumba12) [inline src/core/behavior/shortcut.js]: can u add here another control values like ctrl F1 etc.?
and please add argument isControl(code),
Thanks!

File: src/core/behavior/shortcut.manager.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
+import {flatten, isFunction} from '../utility';
+import {Command} from './command';
+
+export class ShortcutManager {
+	constructor() {
+		this.managerMap = new Map();
+		this.keyCode = '';
```
Comment (klumba12) [inline src/core/behavior/shortcut.manager.js]: Why we need to store here keyCode, looks wrong place here

File: src/core/edit/edit.cell.view.js [hunk 9]
```diff
@@ -49,6 +52,11 @@ export class EditCellView {
 					}
 					else {
 						cell = model.navigation().cell;
+						const code = this.shortcut.keyCode;
+						if (cell && !this.shortcut.isControl) {
+							this.memory = code;
```
Comment (klumba12) [inline src/core/edit/edit.cell.view.js]: memory? for what we need to have it?

File: src/core/edit/edit.cell.view.js [hunk 10]
```diff
@@ -49,6 +52,11 @@ export class EditCellView {
...
+							cell.value += code;
```
Comment (klumba12) [inline src/core/edit/edit.cell.view.js]: i think it should assign code not append like in excel

File: src/core/edit/edit.cell.view.js [hunk 11]
```diff
@@ -109,7 +117,16 @@ export class EditCellView {
 					}
 
 					cell = cell || this.editor.cell || model.navigation().cell;
-					if (cell && model.edit().cancel.execute(this.contextFactory(cell, this.value, this.label)) !== false) {
+					let label = this.label;
```
Comment (klumba12) [inline src/core/edit/edit.cell.view.js:121]: don't understand this code, why we need to anything on cancel? it should just close editor

File: src/core/edit/edit.cell.view.js (hunk 2 above)
Comment (klumba12) [reply src/core/edit/edit.cell.view.js]: why u need to know it ? it's not allowed to cell.value until commit is triggred, for that this.value was introduced

Comment (etokareva) [reply src/core/edit/edit.cell.view.js]: Because I have to add and remove letter, which opens edit-mode

**Review by klumba12 — APPROVED** [review]

**Pull request merged**